docs/overview/architecture.md

```diff
@@ -51,135 +72,33 @@ Here are the concepts of Vald.
-
-1. Vald Ingress receives the delete request from the user. The request includes the vector ID(s), which is specified by the user.
-2. Vald Ingress will forward the request to the Vald Filter Gateway.
-3. Vald Filter Gateway will forward the request to the Vald Meta Gateway. Vald Meta Gateway is used to resolve the internal used UUID of the vector from the Vald Meta.
-4. Vald Meta Gateway will forward the request to the Vald Meta to confirm whether the metadata, which contains the request ID(s), exists or not.
-5. Vald Meta gets the UUID(s) by the request ID(s). Return error if no UUID(s) is found, return with the error.
-6. If Vald Meta Gateway gets the UUID(s), Vald Meta Gateway will forward it to the request with the UUID(s) to the Vald Backup Gateway.
-7. Vale Backup Gateway will forward the request with the UUID(s) to the Vald LB Gateway.
-8. Vald LB Gateway will broadcast the request with UUID(s) to the Vald Agents. Each Vald Agent will delete the vector data and the metadata if the corresponding UUID(s) is found in the in-memory graph index.
-9. If Vald Agent successfully deletes the request data, it will return success to the Vald LB Gateway.
-10. After Vald LB Gateway receives success with location info (e.g, IP address of pod) from the Vald Agent, Vald LB Gateway will return success to the Vald Backup Gateway.
-11. Vald Backup Gateway will forward the request with the UUID to the Vald Compressor.
-12. Vald Compressor will forward the UUID(s) to the Vald Backup Manager.
-13. Vald Backup Manager will delete the data with the same UUID(s).
-14. Vald Backup Gateway will return success to the Vald Meta Gateway if the data is successfully deleted from the Vald Backup Manager.
-15. Vald Meta Gateway will forward the UUID(s) to the Vald Meta.
-16. Vald Meta will delete the data from the persistent layer if it is successfully deleted from the Vald Agents store the deleted UUID(s) and the deletion time in the persistent layer.
-17. Vald Meta Gateway will return success to the Vald Filter Gateway.
-18. Vald Filter Gateway will return success to the Vald Ingress.
+
+## Basic Architecture
+
+Vald is based on microservice, which means Vald is composited by multiple components, you can deploy part of the components to your cluster depending on your needs.
+In this section, we will introduce the basic architecture of Vald.
+
+<img src="../../assets/docs/overview/minimal_architecture.png" />
+
+We will introduce each component and why it is needed in Vald.
+
+### Vald Agent
+
+Vald Agent is the core component of Vald, the approximate nearest neighbor search engine, and stores the graph tree construction on memory for indexing the vectors.
+Agent-NGT uses [yahoojapan/NGT](https://github.com/yahoojapan/ngt) as a core library.
+
+### Vald LB Gateway
+
+Vald LB Gateway is a gateway to load balance the user request and forward user request to the Vald-Agent based on the resource usage of the Vald-Agent and the corresponding cluster node.
```

Comment:
Which is correct, `Vald Agent` or `Vald-Agent`?
I think we need to unify the names like any other component.(`Vald LB Gateway`)

```suggestion
Vald LB Gateway is a gateway to load balance the user request and forward user request to the Vald Agent based on the resource usage of the Vald Agent and the corresponding cluster node.
```